### ISSUE TITLE
Upgrade pygeoif

### ISSUE THREAD

**Issue by Jack-Hayes** [uw-cryo/coincident#12]
Coincident version '0.1.dev28+gf5aee55'
Personal conda env on Python 3.12.5

When running `coincident.search.search(dataset='maxar'...`  I got error `ImportError: cannot import name 'shape' from 'pygeoif' (c:\Users\JackE\anaconda3\envs\JackConda\Lib\site-packages\pygeoif\__init__.py)`

Ran a pip upgrade to fix (went from 0.7 -> 1.5)
<details><summary>Show output</summary>

```
Requirement already satisfied: pygeoif in [c:\users\jacke\anaconda3\envs\jackconda\lib\site-packages](file:///C:/users/jacke/anaconda3/envs/jackconda/lib/site-packages) (0.7)
Collecting pygeoif
  Using cached pygeoif-1.5.0-py3-none-any.whl.metadata (15 kB)
Requirement already satisfied: typing-extensions in [c:\users\jacke\anaconda3\envs\jackconda\lib\site-packages](file:///C:/users/jacke/anaconda3/envs/jackconda/lib/site-packages) (from pygeoif) (4.12.2)
Using cached pygeoif-1.5.0-py3-none-any.whl (27 kB)
Installing collected packages: pygeoif
  Attempting uninstall: pygeoif
    Found existing installation: pygeoif 0.7
    Uninstalling pygeoif-0.7:
      Successfully uninstalled pygeoif-0.7
Successfully installed pygeoif-1.5.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
fastkml 0.12 requires pygeoif<1.0, but you have pygeoif 1.5.0 which is incompatible.
```

</details>

That being said, I get this long output when I run the Maxar dataset search (results returned were expected, though):
<details><summary>Show output</summary>

```
[autoreload of pygeoif.geometry failed: Traceback (most recent call last):
  File "c:\Users\JackE\anaconda3\envs\JackConda\Lib\site-packages\IPython\extensions\autoreload.py", line 276, in check
    superreload(m, reload, self.old_objects)
  File "c:\Users\JackE\anaconda3\envs\JackConda\Lib\site-packages\IPython\extensions\autoreload.py", line 500, in superreload
    update_generic(old_obj, new_obj)
  File "c:\Users\JackE\anaconda3\envs\JackConda\Lib\site-packages\IPython\extensions\autoreload.py", line 397, in update_generic
    update(a, b)
  File "c:\Users\JackE\anaconda3\envs\JackConda\Lib\site-packages\IPython\extensions\autoreload.py", line 349, in update_class
    if update_generic(old_obj, new_obj):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\JackE\anaconda3\envs\JackConda\Lib\site-packages\IPython\extensions\autoreload.py", line 397, in update_generic
    update(a, b)
  File "c:\Users\JackE\anaconda3\envs\JackConda\Lib\site-packages\IPython\extensions\autoreload.py", line 371, in update_property
    update_generic(old.fget, new.fget)
  File "c:\Users\JackE\anaconda3\envs\JackConda\Lib\site-packages\IPython\extensions\autoreload.py", line 397, in update_generic
    update(a, b)
  File "c:\Users\JackE\anaconda3\envs\JackConda\Lib\site-packages\IPython\extensions\autoreload.py", line 309, in update_function
    setattr(old, name, getattr(new, name))
ValueError: __geo_interface__() requires a code object with 0 free vars, not 2040109465601
]
[c:\Users\JackE\anaconda3\envs\JackConda\Lib\site-packages\pystac_client\item_search.py:422](file:///C:/Users/JackE/anaconda3/envs/JackConda/Lib/site-packages/pystac_client/item_search.py:422): DoesNotConformTo: Server does not conform to FILTER
  warnings.warn(DoesNotConformTo("FILTER"))
```

</details>

**Comment by scottyhq** [comment]
> When running coincident.search.search(dataset='maxar'... coincident.search.search(dataset='maxar'...

Thanks @Jack-Hayes, in the future, please include a full executable code example. 

Are you trying to pass a pygeoif object to `coincident.search` as in:
https://pystac-client.readthedocs.io/en/stable/tutorials/item-search-intersects.html#Item-Search-with-AOI-as-a-pygeoif-Object  ?

Unlike pystac_client, coincident.search is currently only setup to accept a geoseries or geodataframe as input:

https://github.com/uw-cryo/coincident/blob/dad5b63a2fe431ad2667a54d4a221029cda751de/src/coincident/search/main.py#L19

**Comment by Jack-Hayes** [comment]
Apologies for not including the original code. Looking back at the notebook, here's the code I executed
```
gf_3dep = coincident.search.search(dataset='3dep',
                            datetime=['2019-10-07', '2019-10-08'],                      
)
site_3dep = gf_3dep[gf_3dep.workunit == "CA_YosemiteNP_2019"].reset_index(drop=True)
for i, row in site_3dep.iterrows():
    start = row.start_datetime
    end = row.end_datetime + pd.Timedelta(days=66)
    gf_maxar = coincident.search.search(dataset='maxar',
                                intersects = site_3dep.iloc[[i]].geometry.envelope,
                                datetime=f"{start.strftime('%Y-%m-%d')}/{end.strftime('%Y-%m-%d')}",
                                filter='eo:cloud_cover < 20', 
    )
    break
```

So I believe the input for the search was a GeoDataFrame which is why I was confused. I remember the cell ran without error after running the `pip install` mentioned above - sorry for the confusion and lack of additional information...

**Comment by scottyhq** [comment]
Fixed by #4
